dali/kernels/imgproc/convolution/filter_gpu.cuh

```diff
@@ -0,0 +1,1094 @@
+                                         const ShapeDesc<axes>& sample_shape) const {
+    return in[dot(coords, sample_shape.in_strides)];
+  }
+};
+
+template <typename In, int axes>
+struct InLoaderPad {
+  DALI_HOST_DEV DALI_FORCEINLINE int border_remap(int idx, const ShapeDesc<axes>& sample_shape,
+                                                  int axis) const {
+    (void)sample_shape;
+    (void)axis;
+    return idx;
+  }
+
+  DALI_HOST_DEV DALI_FORCEINLINE int border_remap_innermost(
+      int idx, const ShapeDesc<axes>& sample_shape) const {
+    (void)sample_shape;
+    return idx;
+  }
+
+  DALI_HOST_DEV DALI_FORCEINLINE In load(const In* __restrict__ in, const ivec<axes>& coords,
+                                         const ShapeDesc<axes>& sample_shape) const {
+    if (any_coord(coords < 0) || any_coord(coords >= sample_shape.in_extents)) {
+      return fill_value;
+    }
+    return in[dot(coords, sample_shape.in_strides)];
+  }
+
+  In fill_value;
+};
+
+template <typename InLoader>
+struct InLoaderFactory {
+  DALI_DEVICE DALI_FORCEINLINE InLoader operator()(int sample_idx) {
+    return {};
+  }
+};
+
+template <typename In, int axes>
+struct InLoaderFactory<InLoaderPad<In, axes>> {
+  DALI_DEVICE DALI_FORCEINLINE InLoaderPad<In, axes> operator()(int sample_idx) {
+    if (fill_values == nullptr) {
+      return {0};
+    }
+    return {fill_values[sample_idx][0]};
+  }
+  const In** fill_values;
+};
+
+/** @} */  // end of InputLoader
+
+
+/** @defgroup InputConv The specializations provide ``compute`` method that computes convolution
+ * for the patch of logical block size (log_block_extents) and stores it in the provided ``acc``.
+ * @{
+ */
+
+/**
+ * @brief First loads the input patch necessary to compute the output of logical block size
+ * (log_block_extents) into shared memory (including filter's halo/apron), then computes the
+ * convolution.
+ */
+template <int lanes_axis_, typename SampleDescT_, typename Inloader_, typename BlockSetupT_>
+struct ShmInputConv {
+  using SampleDescT = SampleDescT_;
+  using Inloader = Inloader_;
+  using BlockSetupT = BlockSetupT_;
+  using StaticConfigT = typename BlockSetupT::StaticConfigT;
+  using In = typename SampleDescT::In;
+  using Acc = typename SampleDescT::Acc;
+  static constexpr int lanes_axis = lanes_axis_;
+
+  DALI_DEVICE DALI_FORCEINLINE void compute(Acc* __restrict__ acc, const In* __restrict__ in,
+                                            const ivec<SampleDescT::axes>& start) const {
+    auto anchored_start = start - sample_desc.shape.anchor_shift;
+    __syncthreads();
+    precompute_indices(in, anchored_start);
+    __syncthreads();
+    load_input_to_shm(in, anchored_start);
+    __syncthreads();
+    const auto& thread_idx = block_setup.thread_idx();
+    stride_filter(sample_desc.shape.filter_extents, [&](auto filter_coef, const auto& filter_offset,
+                                                        int lane) {
```

Comment:
It's not `filter_offset`, because we're not scanning the positions _of_ the filter. We're scanning positions _within_ filter.
It's not exactly input_offset, either, so maybe let's leave it vague (`offset`) rather than outright misleading.
```suggestion
    stride_filter(sample_desc.shape.filter_extents, [&](auto filter_coef, const auto& offset,
                                                        int lane) {
```

Reply:
done